### PR TITLE
Add max of 50 to computed agents. 

### PR DESCRIPTION
**Pull request by cfkoehler**
The configurable property can override this.

### CODE REVIEW

**Review by jpdahlke — CHANGES_REQUESTED** [review]
Please add a unit test. Recommend possibly creating a getMaxMemory() method to leverage/override in test to provide maxMem value rather than messing with the Runtime.

**Comment by cfkoehler** [comment]
Added a test class and test's for the computePoolSize. Change it to take in a parameter of max memory.

**Review by jdcove2 — APPROVED** [review]

File: src/main/java/emissary/pool/AgentPool.java [hunk 1]
```diff
@@ -35,35 +39,40 @@
...
-        // sizePerAgent = conf.findIntEntry("agent.average_size_kb", 1024) * 1024;
-        // }
-        // catch (IOException ex) {
-        // logger.info("Cannot read config file " + ex.getMessage() + ", using default agent size");
-        // }
-
-        long maxMem = Runtime.getRuntime().maxMemory();
-        // UNUSED
-        // float headRoom = 0.40f; // space for places
+    protected static int computePoolSize(final long maxMemoryInBytes, final Integer propertyOverride) {
```
Comment (jpdahlke) [inline src/main/java/emissary/pool/AgentPool.java:46]: Does poolSizeOverride make more sense?

File: src/test/java/emissary/pool/AgentPoolTest.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+    @Test
+    public void testComputePoolSize() throws IllegalArgumentException {
+        // setup expected exception
+        if (expectedException != null) {
+            thrown.expect(expectedException);
+            thrown.expectMessage(expectedExceptionMsg);
+        }
+
+        assertEquals(expectedPoolSize, AgentPool.computePoolSize(maxMemoryInBytes, propertyOverride), "Pool Size Calculation is not as");
```
Comment (jpdahlke) [inline src/test/java/emissary/pool/AgentPoolTest.java:54]: This looks like:
`
Pool Size Calculation is not as ==> expected: <expected val> but was: <actual val>
`
Is this what was desired? "Pool Size Calculation Error"?
`
Pool Size Calculation Error ==> expected: <expected val> but was: <actual val>
`

**Review by jpdahlke — APPROVED** [review]

**Pull request merged**